miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/util/BulkQcTable.java

```diff
@@ -0,0 +1,47 @@
+package uk.ac.bbsrc.tgac.miso.webapp.util;
+
+import java.io.IOException;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.ui.ModelMap;
+import org.springframework.web.servlet.ModelAndView;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+
+import uk.ac.bbsrc.tgac.miso.core.util.WhineyFunction;
+import uk.ac.bbsrc.tgac.miso.dto.QcDto;
+import uk.ac.bbsrc.tgac.miso.service.QualityControlService;
+
+public abstract class BulkQcTable extends BulkTableBackend<QcDto> {
+  private static final Logger log = LoggerFactory.getLogger(BulkQcTable.class);
+  private final boolean create;
+  protected final QualityControlService qcService;
+  protected final String qcTarget;
+  private final String verb;
+
+  public BulkQcTable(String qcTarget, boolean create, QualityControlService qcService, String verb) {
+    super("qc('" + qcTarget + "')", QcDto.class);
+    this.qcTarget = qcTarget;
+    this.create = create;
+    this.qcService = qcService;
+    this.verb = verb;
+  }
+
+  public final ModelAndView display(String idString, ModelMap model) throws IOException {
+
+    return prepare(model, create, verb + " " + qcTarget + " QCs",
+        parseIds(idString).stream().flatMap(WhineyFunction.log(log, this::load))
```

Comment:
I'm really not a fan of the `WhineyFunction` stuff. Here if we ask to create/edit 96 things and one fails to load for any reason, we get no indication and probably don't even notice that there are only 95 rows in the table we now have to work with. Or we ask for 1 thing and this method acts like it was successful after creating a table with no rows?

Reply:
Fixed `WhineyFunction`.